lib/matplotlib/backend_bases.py

```diff
@@ -3543,6 +3525,7 @@ def _trigger_tool(self, name, sender=None, canvasevent=None, data=None):
         # Important!!!
         # This is where the Tool object gets triggered
         tool.trigger(sender, canvasevent, data)
+        tool.state_changed_event(sender, canvasevent, data)
```

Comment:
I would love to, as that will bring it directly to the point it gets fired, however it makes the code more cumbersome to write, with every tool needing a call to super...
```python
def trigger(self, sender, event, data=None):
    ToolBase.trigger(self. sender, event, data)
```
In my old code base (from the summer before my computer died), I did it like this:
```python
def _trigger(*args, **kwargs):
    self.trigger(*args, **kwargs)
    tool.state_changed_event(...)
```
But I didn't like that because it makes trigger a private method...

I guess I could do it the other way around, reversing those methods, make the developer override ``_trigger`` which executes the tool specific code, and have trigger() as the entry to the tool which ensures that the event gets fired.